src/ShellWords.hs

```diff
@@ -28,50 +26,34 @@ parse :: String -> Either String [String]
 parse = runParser parser
 
 runParser :: Parser a -> String -> Either String a
-runParser p = first errorBundlePretty . Megaparsec.parse p "<input>"
+runParser p = first errorBundlePretty . Megaparsec.parse (p <* eof) "<input>"
```

Comment:
Technically, this is a behavior change that could warrant major -- but I'm going to consider it a bug-fix as it was always the intention it should behave this way.